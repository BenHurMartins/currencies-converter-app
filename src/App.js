import React, { useEffect, useState } from 'react';
import './styles.css';
import { getCryptoCurrencies, getCurrencies, } from "./api";
import CurrenciesConverter from "./components/CurrenciesConverter";
import CryptoCurrenciesConverter from "./components/CryptoCurrenciesConverter";
import { i18n } from "./i18n";


function App() {

  const [lang, setLang] = useState('EN')
  //default value
  const [currenciesArray, setCurrenciesArray] = useState([])
  const [currencies, setCurrencies] = useState({})
  //crypto
  const [cryptoCurrenciesArray, setCryptoCurrenciesArray] = useState([])
  const [cryptos, setCryptos] = useState([])

  useEffect(async () => {
    getCurrencies()
      .then(data => {
        setCurrenciesArray(Object.keys(data))
        setCurrencies(data)
      })
      .catch(err => {
        alert("Error trying to get the currencies, try again later")
      })

  }, [])

  useEffect(async () => {
    getCryptoCurrencies()
      .then(data => {
        let cryptoValues = {}
        const dataCryptosSimbols = data.map(item => {
          cryptoValues[item.symbol] = item.name
          return item.symbol
        })
        setCryptoCurrenciesArray(dataCryptosSimbols)
        setCryptos(cryptoValues)
      })
      .catch(err => {
        alert("Error: " + err.error_message)
      })
  }, [])

  return (
    <div className="app">
      <div className="i18n">
        <button className={lang == 'EN' ? "langSelected" : null} onClick={() => setLang('EN')}>EN</button>
        <button className={lang == 'PT_BR' ? "langSelected" : null} onClick={() => setLang('PT_BR')} >PT</button>
      </div>
      <header className="app-header">
        <p className="app-title">
          {i18n.title[lang]}
        </p>
      </header>
      <body className="app-body">
        <CurrenciesConverter
          currenciesArray={currenciesArray}
          currencies={currencies}
          lang={lang}
        />
        <CryptoCurrenciesConverter
          cryptoCurrenciesArray={cryptoCurrenciesArray}
          currenciesArray={currenciesArray}
          cryptos={cryptos}
          currencies={currencies}
          lang={lang}
        />
      </body>
    </div>
  );
}

export default App;
