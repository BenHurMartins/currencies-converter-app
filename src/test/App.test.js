import React from 'react';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from "../App";
afterEach(cleanup);

test('Renders App.js correctly', () => {
  const component = renderer.create(<App />)

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('Should Start the component with english language selected', () => {
  const { getByText } = render(
    <App />,
  );

  expect(getByText(/Currency Converter/i)).toBeTruthy();

})
it('Should change the text after another language is selected', () => {
  const { getByText, getAllByText } = render(
    <App />,
  );

  expect(getAllByText(/PT/i)[0]).toBeTruthy();

  fireEvent.click(getAllByText(/PT/i)[0])


  expect(getByText(/Conversor de Moedas/i)).toBeTruthy();


})