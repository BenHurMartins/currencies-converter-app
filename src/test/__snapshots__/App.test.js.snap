// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders App.js correctly 1`] = `
<div
  className="app"
>
  <div
    className="i18n"
  >
    <button
      className="langSelected"
      onClick={[Function]}
    >
      EN
    </button>
    <button
      className={null}
      onClick={[Function]}
    >
      PT
    </button>
  </div>
  <header
    className="app-header"
  >
    <p
      className="app-title"
    >
      Currency Converter
    </p>
  </header>
  <body
    className="app-body"
  >
    <p
      className="app-subtitle"
    >
      Currency
    </p>
    <div
      className="main-container"
    >
      <div
        className="form-container"
      >
        <div
          className="half-container"
        >
          <p
            className="label"
          >
            From:
          </p>
          <input
            min="0"
            onChange={[Function]}
            pattern="^\\\\d+(?:\\\\.\\\\d{1,2})?$"
            placeholder="0.00"
            step="0.01"
            title="Currency"
            value={0}
          />
          <select
            onChange={[Function]}
            value="USD"
          />
        </div>
        <div
          className="half-container"
        >
          <p
            className="label"
          >
            To:
          </p>
          <input
            placeholder="0.00"
            value={0}
          />
          <select
            onChange={[Function]}
            value="EUR"
          />
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Convert
        </button>
      </div>
    </div>
    <p
      className="app-subtitle"
    >
      Crypto Currency
    </p>
    <div
      className="main-container"
    >
      <div
        className="form-container"
      >
        <div
          className="half-container"
        >
          <p
            className="label"
          >
            From:
          </p>
          <input
            className="input-from"
            min="0"
            onChange={[Function]}
            placeholder="0.00"
            value={0}
          />
          <select
            onChange={[Function]}
            value="BTC"
          />
        </div>
        <div
          className="half-container"
        >
          <p
            className="label"
          >
            To:
          </p>
          <input
            className="input-to"
            placeholder="0.00"
            value={0}
          />
          <select
            onChange={[Function]}
            value="USD"
          />
        </div>
      </div>
      <div>
        <button
          onClick={[Function]}
        >
          Convert Crypto
        </button>
      </div>
    </div>
  </body>
</div>
`;
